README.md
# PIIP
Kids educational game


changelog.txt
Week 4, Monday, 25/March
Finished
- Read the assessment booklet for the structure requirements of project
- Created a basic structure of project folder with all necessary folders and files.
- Created the index.html and filled it with basic html structure
- Started working on data collection and structuring
- Formed items.js and rooms.js for holding the basic data.
- Included the files in index.html and checked that they are loading in browser
Still working on
- Collect images required.
- Form separate data js file for rooms.
Milestone 1 progress
- The project is kicked off with just a 2% progress this week.
- Target to complete the data collection to proceed with coding.

Week 4, Tuesday, 26/March
Finished
- Started collecting images of various rooms
- Filled in license of images accordingly
- Filled in rooms.js with data and images of five rooms
Still working on
- Collect images of items.
- Form separate js file for items.
Milestone 1 progress
- Data collection is still in progress and a 5% progress overall is achieved.
- Complete the data for images as well and get done with data collection.

Week 4, Wednesday, 27/March
Finished
- Started collecting images of various items in each room.
- Populated the license accordingly
- Finished collecting details of items of two rooms.
- Filled in the data to items.js
Still working on
- Finish collecting images for all the rooms.
- Populate and finalize the items and rooms.js
Milestone 1 progress
- The data collection is still in progress.

Week 4, Thursday, 28/March
Finished
- Collected more images for remaining three rooms
- A total of 32 images are collected and entered into license text file.
- rooms.js and items.js is ready with necessary data.
Still working on
- Fonts download.
- Style sheet creation and font inclusion.
Milestone 1 progress
- Data collection part is almost over which constitutes to 10% completion.
- Target to set up the initial set up viz fonts, sounds and necessary
   style sheets.

Week 4, Friday, 29/March
Finished
- Downloaded the sansserifpro fonts from font squirrel
- Included the fonts into the css and defined a font family
- Included styling to include the font for the html body
- Included the style sheet game.css into HTML
- Tested for the font to appear on browser
Still working on
- Download different sounds required for the game.
- Create various js files required for the game.
Milestone 1 progress
- The initial set up is still in progress. A 15% completion can be marked.
- Target to complete all the external downloads required.

Week 4, Saturday, 30/March
Finished
- Searched for sounds for different pages
- Downloaded the background sounds.
- Respective license is recorded.
Still working on
- Create the js files for various functionalities.
Milestone 1 progress
- The initial set up is still in progress. The progress rate is at 18%.
- Target to complete the js set up and start with initial page coding.

Week 4, Sunday, 31/March
Finished
- Wrote a onLoad function in app.js for initiating game landing page
- Wrote a function in ui.js to add game title
- Wrote a function in ui.js to add buttons
Still working on
- Create separate js for game elements.
- Create content for landing page.
- Create a on load function for loading game.
Milestone 1 progress
- The initial set up is finally done marking a 20% of progress.
- Target to complete the components of landing page and finally the game
  pages for release one.

Week 5, Monday, 1/April
Finished
- Created the file gameElements.js for holding creating elements
- Created function createLandingPageContent for landing page content
- Created function loadLandingPage for loading the first page
- Added code in load game to utilize the above functions and load the pages
Still working on
- Add styles for landing page
Milestone 1 progress
- Completed about 25% of the targets for release one.
- Target to complete the landing page and proceed with level selection page.

Week 5, Tuesday, 2/April
Finished
- Added the css for the landing page in game.css
- Modified code in landing page functions to add the classes accordingly
Still working on
- Add functions for level selection page
- Add styles for the level selection page
Milestone 1 progress
- One of the page is completed. Around 30% of progress is achieved.
- Target to comaplete the level selection page and progress to the main game
  page

Week 5, Wednesday, 3/April
Finished
- Added functions for level selection page which include
  createLevelSelectionPage, loadLevelSelectionPage
- Added css for the level selection page
Still working on
- Create header for game page.
- Add event handlers to load game as per level button clicked.
Milestone 1 progress
- Completed about 40%of targets.
- Target to start with the game page.

Week 5, Thursday, 4/April
Finished
- Added the event handlers for level selection buttons in level selection
  page which includes the functions addLevelEventHandler, loadGamePage,
  loadGamePage
- Created function createGameLevelHeader for the header part of game.
Still working on
- Create the left and right panels of the game.
- Create logic to randomly fetch rooms and items.
Milestone 1 progress
- Two of the pages targeted for release 1 are done marking 50% of
  progress.
- Target to complete the game page with necessary logic in place.

Week 5, Friday, 5/April
Finished
- Added the function createLeftPanel for the left part of game
- Added the file utilityFunctions.js with the getRandomInt function to
  get random number between two digits
- Added function fetchRoomList to select random rooms
Still working on
- Create right panel with the question element.
- Fetch item as per the rooms randomly selected in left panel.
Milestone 1 progress
- Completed around 60% of target to be achieved.
- Target to complete the right panel with main logic of drag and drop

Week 5, Saturday, 6/April
Finished
- Added the CSS for the left panel of the game
- Added the function createRightPanel for the right side part of game

data/items.js
/*
  items.js: Creates global variable 'items' that Contains
  list of different javascript item objects

  items : List og javascript objects
  
  item object:
  itemId: unique to a particular item
  itemName: Name to be dispalyed/checked for the item
  imageUrl: url for the image to be displayed for the item
*/
window.items = [
  {
    itemId: 1,
    itemName: "Fridge",
    imageUrl: "images/fridge.svg",
  },
  {
    itemId: 2,
    itemName: "Knife",
    imageUrl: "images/knife.svg",
  },
  {
    itemId: 3,
    itemName: "Cooking Pot",
    imageUrl: "images/cooking_pot.svg",
  },
  {
    itemId: 4,
    itemName: "Toaster",
    imageUrl: "images/toaster.svg",
  },
  {
    itemId: 5,
    itemName: "Coffee Maker",
    imageUrl: "images/coffee_maker.svg",
  },
  {
    itemId: 6,
    itemName: "Bed",
    imageUrl: "images/bed.svg"
  },
  {
    itemId: 7,
    itemName: "Wardrobe",
    imageUrl: "images/wardrobe.svg"
  },
  {
    itemId: 8,
    itemName: "Dresser",
    imageUrl: "images/dresser.svg"
  },
  {
    itemId: 9,
    itemName: "Bunk bed",
    imageUrl: "images/bunkbed.svg"
  },
  {
    itemId: 10,
    itemName: "Bed lamp",
    imageUrl: "images/bedlamp.svg"
  },
  {
    itemId: 11,
    itemName: "Couch",
    imageUrl: "images/couch.svg"
  },
  {
    itemId: 12,
    itemName: "Table",
    imageUrl: "images/table.svg"
  },
  {
    itemId: 13,
    itemName: "Dining table",
    imageUrl: "images/dining_table.svg"
  },
  {
    itemId: 14,
    itemName: "Television",
    imageUrl: "images/television.svg"
  },
  {
    itemId: 15,
    itemName: "Fireplace Mantle",
    imageUrl: "images/fireplace.svg"
  },
  {
    itemId: 16,
    itemName: "Toilet paper",
    imageUrl: "images/toilet_paper.svg"
  },
  {
    itemId: 17,
    itemName: "Toilet basin",
    imageUrl: "images/toilet_basin.svg"
  },
  {
    itemId: 18,
    itemName: "Wash basin",
    imageUrl: "images/wash_basin.svg"
  },
  {
    itemId: 19,
    itemName: "Bath tub",
    imageUrl: "images/bath_tub.svg"
  },
  {
    itemId: 20,
    itemName: "Toothbrush",
    imageUrl: "images/tooth_brush.svg"
  },
  {
    itemId: 21,
    itemName: "Shovel",
    imageUrl: "images/shovel.svg"
  },
  {
    itemId: 22,
    itemName: "Watering can",
    imageUrl: "images/watering_can.svg"
  },
  {
    itemId: 23,
    itemName: "Gloves",
    imageUrl: "images/gloves.svg"
  },
  {
    itemId: 24,
    itemName: "Gardening rake",
    imageUrl: "images/rake.svg"
  },
  {
    itemId: 25,
    itemName: "Wheelbarrow",
    imageUrl: "images/wheelbarrow.svg"
  },
  {
    itemId: 26,
    itemName: "Microwave oven",
    imageUrl: 'images/microwaveoven.svg'
  },
  {
    itemId: 27,
    itemName: 'Mincer',
    imageUrl: 'images/mincer.svg'
  },
  {
    itemId: 28,
    itemName: 'stove',
    imageUrl: 'images/stove.svg'
  },
  {
    itemId: 29,
    itemName: 'kettle',
    imageUrl: 'images/kettle.svg'
  },
  {
    itemId: 30,
    itemName: 'blender',
    imageUrl: 'images/blender.svg'
  },
  {
    itemId: 31,
    itemName: 'pillows',
    imageUrl: 'images/pillows.svg'
  },
  {
    itemId: 32,
    itemName:'Mattress',
    imageUrl: 'images/mattress.svg'
  },
  {
    itemId: 33,
    itemName: 'Alarm',
    imageUrl: 'images/alarm.svg'
  },
  {
    itemId: 34,
    itemName:'Cradle',
    imageUrl: 'images/cradle.svg'
  },
  {
    itemId: 35,
    itemName: 'Blanket',
    imageUrl: 'images/blanket.svg'
  },
  {
    itemId: 36,
    itemName: 'Armchair',
    imageUrl: 'images/armchair.svg'
  },
  {
    itemId:37,
    itemName: 'Carpet',
    imageUrl: 'images/carpet.svg'
  },
  {
    itemId: 38,
    itemName: 'Speaker',
    imageUrl: 'images/speaker.svg'
  },
  {
    itemId: 39,
    itemName: 'Curtains',
    imageUrl: 'images/curtains.svg'
  },
  {
    itemId: 40,
    itemName: 'Bean bag',
    imageUrl: 'images/bean_bag.svg'
  },
  {
    itemId: 41,
    itemName: 'Dispenser',
    imageUrl: 'images/dispenser.svg'
  },
  {
    itemId: 42,
    itemName: 'Towel',
    imageUrl: 'images/towel.svg'
  },
  {
    itemId: 43,
    itemName: 'Shower',
    imageUrl: 'images/shower.svg'
  },
  {
    itemId: 44,
    itemName: 'Hand Dryer',
    imageUrl: 'images/hand_dryer.svg'
  },
  {
    itemId: 45,
    itemName: 'Hair Dresser',
    imageUrl: 'images/hair_dresser.svg'
  },
  {
    itemId: 46,
    itemName: 'Bird House',
    imageUrl: 'images/birdhouse.svg'
  },
  {
    itemId: 47,
    itemName: 'Garden table',
    imageUrl: 'images/garden_table.svg'
  },
  {
    itemId: 48,
    itemName: 'Bush',
    imageUrl: 'images/bush.svg'
  },
  {
    itemId: 49,
    itemName: 'Axe',
    imageUrl: 'images/axe.svg'
  },
  {
    itemId: 50,
    itemName: 'Lawnmower',
    imageUrl: 'images/lawnmower.svg'
  }
]
